Add warning when pg:copy-ing from db with additional credentials

camillebaldock opened the pull request:
Part a) of https://trello.com/c/htDenfnh/2182-pgcopy-copyover-does-not-work-with-additional-creds

camillebaldock committed:
Add warning when pg:copy-ing from db with additional credentials

msakrejda commented on commands/copy.js:
```diff
@@ -51,6 +51,12 @@ Data from ${cli.color.yellow(source.name)} will then be transferred to ${cli.col
+      from_url: source.url,
+      to_name: target.name,
+      to_url: target.url
+    })
```
Stray debugging?

msakrejda reviewed (changes requested):
Some linting issues, but otherwise looks good.

keiko713 commented on commands/copy.js:
```diff
@@ -61,6 +61,14 @@ Data from ${cli.color.yellow(source.name)} will then be transferred to ${cli.col
   }))
+
+  if (source.attachment) {
+    let credentials = yield heroku.get(`/postgres/v0/databases/${source.attachment.addon.name}/credentials`,
```
let's only do this when it's shogun resource (aka this won't work for yobuko)

msakrejda replied:
This should work for Yobuko as well--we shipped a similar degenerate credentials API there. But if we know the plan up-front, maybe we should avoid this check, since hobby databases will never have additional credentials? I'm +0 on that.

keiko713 replied:
GUESS WHO MADE IT https://github.com/heroku/yobuko/commits/78c42881132e9c9294cc0cb2b20b036463095f7b/web/endpoints/client/postgres.rb 🤦‍♂️  I should have asked Keiko 2017.

Well in that case, yeah I'm +0 as well, it's okay to ship as is.

keiko713 commented:
one comment but looks good, did you test in staging?

camillebaldock commented:
@keiko713 i tested against two databases doing pg:copy before i added a credential, pg:copy after i added a credential (and also tested what happens when you remove a credential, which it turns out takes a long time, and i found a bug which i am opening a PR in shogun for)

camillebaldock commented:
@keiko713 going to test on some yobukos now

camillebaldock commented:
@keiko713 tested with yobuko, behaves as expected (no difference to current behaviour for yobukos)

camillebaldock commented:
@uhoh-itsmaciek are you happy with these changes now?

msakrejda reviewed (approved):
Yes!
